### DISCUSSION TITLE
Ctl File

### DISCUSSION THREAD

**Discussion by ashi-mark** [bhoogter/VB6TocSharp#7] [General]
Good Job
Does the software know how to work on Ctl files?
Thanks

**Comment by bhoogter** [comment]
Usercontrols...  They're not currently supported, although I don't think it would take a whole lot of re-work to get going...

With usercontrols, there are a few options, at least... Since you're limited to a 32-bit exe most likely anyway (due to the PowerPacks), you could just compile the CTLs to the binary OCX files with a custom project.  This would ensure 100% compatibility.  Or, if you want to preserve most of the business logic, for now, you could copy the `.ctl` file to a `.bas` file, and just convert it that way, just to get the code transformed.

I'll look into more automated conversion of the ctl files.. I threw together a branch on this repo just now w/ about 30 mins of work... it allows the processing of ctls directly from the UI (or immediate window), but it'll take just a little bit more to convert the UI elements as well (it's treating it like a module).  I don't know what type of conversion it'll do, but it'd be better than coding from scratch if there's more than 10 or so elements on the UC.

You can either try one of the above, or the branch I'm working with, or check back in a few days and the UI will probably handle it by then.

Ben
